crates/ruff/resources/test/fixtures/pyflakes/F821_17.py

```diff
@@ -0,0 +1,102 @@
+"""Test type parameters and aliases"""
+
+# Type parameters in type alias statements
+
+from some_module import Bar
+
+type Foo[T] = T # OK
+type Foo[T] = list[T] # OK
+type Foo[T: Forward] = T # OK
+type Foo[*Ts] = Bar[Ts] # OK
+type Foo[**P] = Bar[P] # OK
+
+# Types used in aliased assignment must exist
+
+type Foo = DoesNotExist  # F821: Undefined name `DoesNotExist`
+type Foo = list[DoesNotExist]  # F821: Undefined name `DoesNotExist`
+
+# Type parameters do not escape alias scopes
+
+type Foo[T] = T
+T  # F821: Undefined name `T` - not accessible afterward alias scope
+
+# Type parameters in functions
+
+def foo[T](t: T) -> T: return t  # OK
+async def afoo[T](t: T) -> T: return t  # OK
+def with_forward_ref[T: Forward](t: T) -> T: return t  # OK
+def can_access_inside[T](t: T) -> T:  # OK
+    print(T)  # OK
+    return t  # OK
+class Forward: ...  # OK
```

Comment:
Self: Need to move `Forward` definition to end of file or change names so they do not collide across test cases.

Reply:
Another benefit of isolating fixtures >.<